Lib/ntpath.py

```diff
@@ -58,8 +58,31 @@ def normcase(s):
 # volume), or if a pathname after the volume-letter-and-colon or UNC-resource
 # starts with a slash or backslash.
 
+try:
+    from nt import _path_splitroot
+except ImportError:
+    _path_splitroot = _splitroot = None
+else:
+    def _splitroot(p):
+        p = os.fspath(p)
+        r = _path_splitroot(p)
+        if isinstance(p, bytes):
+            return tuple(map(os.fsencode, r))
+        return r
```

Comment:
Maybe change the name to `nt._path_splitdrive`, since that makes more sense now that the root is shifted over to the path side. Implement the POSIX fallback as `_path_splitdrive`, which only returns `str`, just like the builtin function. Move the above bytes support code into a common implementation of `ntpath.splitdrive()`, which is likely the only place it will ever be needed. `_path_splitdrive()` will be used directly when bytes support isn't required.

I'm sure you know the `os.fspath()` call isn't necessary since it's handled for a `path_t` argument in C, so why is it called explicitly here?

Reply:
The `isinstance` at 69 needs to recognise values for `p` that have `__fspath__` that returns bytes, as well as actual bytes, which is why we `fspath` once instead of doing it multiple times.

Reply:
> The `isinstance` at 69 needs to recognise values for `p` that have `__fspath__` that returns bytes, as well as actual bytes, which is why we `fspath` once instead of doing it multiple times.

Sigh... sorry about that. :-$